src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala

```diff
@@ -40,6 +40,15 @@ object DatabaseConversions {
     case _ => None
   }
 
+  //Note, cycle 0 starts at the level 2 block
+  def extractCycle(block: Block) : Option[Int] = {
```

Comment:
Alternative using the existing partial function `discardGenesis` would be

```scala
def extractCycle(block: Block): Option[Int] =
  discardGenesis.lift(block.data.metadata) //this returns an Option[BlockHeaderMetadata]
    .map(_.level.cycle) //this is Option[Int]
```

Which might require no additional function but can be inlined at call site too, if willing